src/vscode-dts/vscode.proposed.lmTools.d.ts

```diff
@@ -91,7 +91,27 @@ declare module 'vscode' {
 		 * Invoke a tool with the given parameters.
 		 * TODO@API Could request a set of contentTypes to be returned so they don't all need to be computed?
 		 */
-		export function invokeTool(id: string, parameters: Object, token: CancellationToken): Thenable<LanguageModelToolResult>;
+		export function invokeTool(id: string, options: LanguageModelToolInvokationOptions, token: CancellationToken): Thenable<LanguageModelToolResult>;
+	}
+
+	export interface LanguageModelToolInvokationOptions {
+		/**
+		 * Parameters with which to invoke the tool.
+		 */
+		parameters: Object;
+		/**
+		 * If known, the maximum number of tokens the tool should emit in its result.
+		 * If this is provided, the {@link countTokens} function must also be provided.
+		 */
+		tokenBudget?: number;
+
+		/**
+		 * Count the number of tokens in a message using the model specific tokenizer-logic.
+		 * @param text A string or a message instance.
+		 * @param token Optional cancellation token.  See {@link CancellationTokenSource} for how to create one.
+		 * @returns A thenable that resolves to the number of tokens.
+		 */
+		countTokens?(text: string | LanguageModelChatMessage, token?: CancellationToken): Thenable<number>;
```

Comment:
Any reason to provide this without setting `tokenBudget`? Should they be in some `tokenOptions?: { tokenBudget; countTokens... }`?